src/components/Breadcrumb/Breadcrumb.tsx

```diff
@@ -0,0 +1,123 @@
+ * is found title is not replaced.
+ *
+ * URL manipulation is done on String basis with + or - on the beginning of the string url parameter coming from page
+ * crumb.
+ *
+ * url: '-/edit' - this will take url from match and remove part of the url from specified parameter to the end
+ *  - before: /projects/100/edit
+ *  - after: /projects/100
+ *
+ * url: '+/edit' - this will take url from match and add specified path to existing url
+ *  - before: /projects/100
+ *  - after: /projects/100/edit
+ *
+ * On top of that there is custom=true page crumbs that don't match with match.handle crumbs. These are added in order
+ * like specified on pages after all matched crumbs. Titles are not manipulated and are coming from pages, url is not
+ * present by default because they don't have match from handle crumb. Url could be supplied from last handle crumb by
+ * using url: '+somethingYouWantToAdd' parameter. This will join url from previous crumb with path addition specified
+ * in current crumb.
+ */
+export const Breadcrumb = ({ pageBreadcrumbs }: IBreadcrumbProps) => {
+  const convertedBreadcrumbData = convertBreadcrumbData(pageBreadcrumbs);
+  const pageBreadcrumbsObject = convertedBreadcrumbData.breadcrumbObject;
+
+  const matches = useMatches();
+  const matchCrumbs = matches.filter((match) => !isUndefined(match.handle) && typeof match.handle === 'string');
+
+  const breadcrumbArray = matchCrumbs
+    .map((match) => {
+      const matchHandle = match.handle as TBreadcrumb;
+
+      let breadcrumb: IBreadcrumbData;
+      const pageBreadcrumb = pageBreadcrumbsObject[matchHandle];
+
+      if (!pageBreadcrumb) {
+        //
+        breadcrumb = { entity: matchHandle, title: breadcrumbData[matchHandle].title, url: match.pathname };
+      } else {
+        //
+        let matchPathname = match.pathname;
+        if (pageBreadcrumb.url?.startsWith('+')) {
+          matchPathname = match.pathname + pageBreadcrumb.url.substring(1);
+        } else if (pageBreadcrumb.url?.startsWith('-')) {
+          matchPathname = match.pathname.split(pageBreadcrumb.url.substring(1))[0];
+        }
+
+        const titleHelp = pageBreadcrumb.title
+          ? pageBreadcrumb.title
+          : breadcrumbData[pageBreadcrumb.entity as TBreadcrumb].title;
+        breadcrumb = { entity: matchHandle, title: titleHelp, url: matchPathname };
+      }
+      return breadcrumb;
+    })
+    .concat(convertedBreadcrumbData.breadcrumbCustoms);
+
+  const breadcrumbs = breadcrumbArray.map((breadcrumb, index, array) => {
+    const isLast = array.length - 1 === index;
+    if (breadcrumb) {
+      if (breadcrumb.url?.startsWith('+')) {
+        breadcrumb.url = array[index - 1]?.url + breadcrumb.url.substring(1);
+      }
+      return (
+        <BreadcrumbItem key={index} isActive={isLast}>
+          {isLast ? breadcrumb.title : <Link to={breadcrumb.url!}> {breadcrumb.title}</Link>}
```

Comment:
I think empty space is mistake? ` {breadcrumb.title}`, it should be just `{breadcrumb.title}`